llvm/test/tools/sycl-post-link/emit_exported_symbols.ll

```diff
@@ -0,0 +1,129 @@
+; This test checks that the post-link tool generates list of exported symbols.
+;
+; Global scope
+; RUN: sycl-post-link -symbols -emit-exported-symbols -S %s -o %t.global.files.table
+; RUN: FileCheck %s -input-file=%t.global.files.table --check-prefixes CHECK-GLOBAL-TABLE
+; RUN: FileCheck %s -input-file=%t.global.files_0.prop --check-prefixes CHECK-GLOBAL-PROP
+;
+; Per-module split
+; RUN: sycl-post-link -symbols -split=source -emit-exported-symbols -S %s -o %t.per_module.files.table
+; RUN: FileCheck %s -input-file=%t.per_module.files.table --check-prefixes CHECK-PERMODULE-TABLE
+; RUN: FileCheck %s -input-file=%t.per_module.files_0.prop --check-prefixes CHECK-PERMODULE-0-PROP
+; RUN: FileCheck %s -input-file=%t.per_module.files_1.prop --check-prefixes CHECK-PERMODULE-1-PROP
+; RUN: FileCheck %s -input-file=%t.per_module.files_2.prop --check-prefixes CHECK-KERNELONLY-PROP
+;
+; Per-kernel split
+; RUN: sycl-post-link -symbols -split=kernel -emit-exported-symbols -S %s -o %t.per_kernel.files.table
+; RUN: FileCheck %s -input-file=%t.per_kernel.files.table --check-prefixes CHECK-PERKERNEL-TABLE
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_0.prop --check-prefixes CHECK-PERKERNEL-0-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_1.prop --check-prefixes CHECK-PERKERNEL-1-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_2.prop --check-prefixes CHECK-PERKERNEL-2-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_3.prop --check-prefixes CHECK-KERNELONLY-PROP
+; RUN: FileCheck %s -input-file=%t.per_kernel.files_4.prop --check-prefixes CHECK-KERNELONLY-PROP
+
+target triple = "spir64-unknown-unknown"
+
+define dso_local spir_kernel void @SpirKernel1(float %arg1) #0 {
+entry:
+  ret void
+}
+
+define dso_local spir_kernel void @SpirKernel2(float %arg1) #2 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @ExportedSpirFunc1(float %arg1) #0 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @ExportedSpirFunc2(i32 %arg1, i32 %arg2) #1 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @ExportedSpirFunc3(float %arg1) #0 {
+entry:
+  ret void
+}
+
+define dso_local spir_func void @NotExportedSpirFunc1(float %arg1) {
+entry:
+  ret void
+}
+
+attributes #0 = { "sycl-module-id"="a.cpp" }
+attributes #1 = { "sycl-module-id"="b.cpp" }
+attributes #2 = { "sycl-module-id"="c.cpp" }
+
+; Global scope
+; CHECK-GLOBAL-PROP: [SYCL/exported symbols]
+; CHECK-GLOBAL-PROP-NEXT: ExportedSpirFunc1
+; CHECK-GLOBAL-PROP-NEXT: ExportedSpirFunc2
+; CHECK-GLOBAL-PROP-NEXT: ExportedSpirFunc3
+; CHECK-GLOBAL-PROP-NOT: SpirKernel1
+; CHECK-GLOBAL-PROP-NOT: NotExportedSpirFunc1
```

Comment:
Alternatively, you can outline those into `--implicit-check-not` directives at each command. With having to check two distinct substrings it might not be as readable as it is now, but if you rename `SpirKernel1` into `NotExportedSpirKernel1`, then `--implicit-check-not` approach could look cleaner

Reply:
Just to make sure I understand, your suggestion is to add `NotExported` as a prefix to the kernel names and then add `--implicit-check-not="NotExported"` to all the file-checks?

Reply:
> Just to make sure I understand, your suggestion is to add NotExported as a prefix to the kernel names and then add --implicit-check-not="NotExported" to all the file-checks?

I'm fine with the current approach as well, but you got the idea correctly - it will save some lines and won't reduce readability (hopefully)

Reply:
I have added the implicit-check-not.